plugins/services/src/js/reducers/serviceForm/Containers.js

```diff
@@ -0,0 +1,156 @@
+import {SET, ADD_ITEM, REMOVE_ITEM} from '../../../../../../src/js/constants/TransactionTypes';
+import Transaction from '../../../../../../src/js/structs/Transaction';
+import {combineReducers, simpleFloatReducer} from '../../../../../../src/js/utils/ReducerUtil';
+
+const containerReducer = combineReducers({
+  cpus: simpleFloatReducer('resources.cpus'),
+  mem: simpleFloatReducer('resources.mem'),
+  disk: simpleFloatReducer('resources.disk')
+});
+
+function containers(state, {type, path = [], value}) {
+  if (!path.includes('containers')) {
+    return state;
+  }
+
+  if (this.cache == null) {
+    this.cache = [];
+  }
+
+  if (!state) {
+    state = [];
+  }
+
+  let newState = state.slice();
+  const index = path[1];
+  const joinedPath = path.join('.');
+
+  if (joinedPath === 'containers') {
+    switch (type) {
+      case ADD_ITEM:
+        newState.push({name: `container${newState.length + 1}`});
```

Comment:
Everywhere in the UI we fallback to `container[space]<index>` let's add space here or remove in the UI 💪